framework/meta/src/tools/wasm_extractor.rs

```diff
@@ -0,0 +1,90 @@
+use colored::Colorize;
+use std::{fs, str::from_utf8};
+use wasmparser::{BinaryReaderError, Parser, Payload};
+
+pub struct WasmInfo {
+    pub imports: Vec<String>,
+    pub allocator_trigger: bool,
+}
+
+impl WasmInfo {
+    const fn new() -> Self {
+        WasmInfo {
+            imports: Vec::new(),
+            allocator_trigger: false,
+        }
+    }
+
+    pub fn set_imports(&mut self, output_wasm_path: &str) {
+        let wasm_data = fs::read(output_wasm_path)
+            .expect("error occured while extracting imports from .wasm: file not found");
+
+        let imports = parse_wasm_imports(wasm_data)
+            .expect("error occured while extracting imports from .wasm ");
+
+        self.imports = imports;
+    }
+
+    pub fn build_wasm_info(output_wasm_path: &str) -> Self {
+        let wasm_data = fs::read(output_wasm_path).expect(
+            "error occured while extracting memory allocation information from .wasm: file not found",
+        );
+
+        let allocator_trigger = is_memory_allocation(wasm_data)
+            .expect("error occured while extracting memory allocation information from .wasm ");
+
+        if allocator_trigger {
+            println!(
+                "{}",
+                "FailAllocator triggered: memory allocation forbidden"
```

Comment:
"FailAllocator used while memory allocation is accessible in code. Contract may fail unexpectedly when memory allocation is attempted".